database/postgresconn/postgresconn.go

```diff
@@ -62,6 +64,10 @@ type Postgres struct {
 	isLocked atomic.Bool
 	// Open and WithConn need to guarantee that config is never nil
 	config *Config
+	// tx transaction surrounding migration
+	tx *sql.Tx
+	// context used during migrations
+	ctx context.Context
```

Comment:
Might be good to also add context aware methods, eg RunCtx. Up to you.

Reply:
Yeah, I have thought about the lack of context aware in these functions. I would come back for that, just because the impact is quite a lot of changes to tests etc.

Reply:
I mean adding entirely new functions that respect context, not updating the old ones

Reply:
Got it. Sounds reasonable